consensus/tool/src/main.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A cli tool for interrogating a mobilecoin consensus node.
+
+#![deny(missing_docs)]
+
+use clap::{Parser, Subcommand};
+use grpcio::{ChannelBuilder, EnvBuilder};
+use mc_common::logger::{create_app_logger, o};
+use mc_consensus_api::{consensus_common_grpc::BlockchainApiClient, empty::Empty};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::ConsensusClientUri;
+use std::{sync::Arc, time::Duration};
+
+/// Command line config, set with defaults that will work with
+/// a standard mobilecoind instance
+#[derive(Clone, Debug, Parser)]
+#[clap(
+    name = "mc-consensus-tool",
+    about = "A cli tool for interrogating one or more consensus nodes"
+)]
+pub struct Config {
+    /// Command to use
+    #[clap(subcommand)]
+    pub tool_command: ToolCommand,
+
+    /// Consensus_uri's to connect to
+    #[clap(global = true)]
+    pub consensus_uris: Vec<ConsensusClientUri>,
+}
+
+/// Commands that the tool recognizes
+#[derive(Clone, Debug, Subcommand)]
+pub enum ToolCommand {
+    /// Status: Prints the network status, including block count and block
+    /// version
+    Status,
+    /// Wait-for-quiet: Blocks until the network is 'quiet'. This means that the
+    /// block count doesn't move for some time.
+    WaitForQuiet {
+        /// Number of seconds the network must stop moving for. Defaults to 20.
+        #[clap(long)]
+        period: Option<u64>,
+    },
+}
+
+const DEFAULT_PERIOD_SECONDS: u64 = 20;
+
+fn main() {
+    let (logger, _global_logger_guard) = create_app_logger(o!());
+    let config = Config::parse();
+
+    let env = Arc::new(EnvBuilder::new().name_prefix("consensus-tool-grpc").build());
+
+    let blockchain_conns: Vec<_> = config
+        .consensus_uris
+        .iter()
+        .map(|uri| {
+            let ch =
+                ChannelBuilder::default_channel_builder(env.clone()).connect_to_uri(uri, &logger);
+            BlockchainApiClient::new(ch)
+        })
+        .collect();
+
+    match config.tool_command {
+        ToolCommand::Status => {
+            for conn in &blockchain_conns {
+                let last_block_info = conn
+                    .get_last_block_info(&Empty::new())
+                    .expect("get last block info");
+                println!("{:?}", last_block_info)
```

Comment:
Printing JSON might be more useful since that will allow the caller to pipe to `jq` and extract specific information, if needed.

Reply:
yeah okay -- I'm not sure if a json version of this struct exists but I can try to make one or do something adhoc I guess

Reply:
Unfortunately the protobuf package v2.27 does not support JSON, even though it's [a well-defined mapping](https://developers.google.com/protocol-buffers/docs/proto3#json). Given the [`impl From<LastBlockInfoResponse> for BlockInfo`](https://github.com/mobilecoinfoundation/mobilecoin/blob/03fdc87f6ed4734a4ca49ccebc28f4ad9f30c5a5/connection/src/traits.rs#L103), I suggest we add `derive(Serialize)` to `BlockInfo`, convert to that, and serialize it with `serde_json`

Reply:
yeah this is the best way, will fix

Reply:
maybe we should move `BlockInfo` to `mc-api` crate